gems/aws-sdk-s3/spec/multipart_stream_uploader_spec.rb

```diff
@@ -0,0 +1,274 @@
+# frozen_string_literal: true
+
+require_relative 'spec_helper'
+require 'tempfile'
+
+module Aws
+  module S3
+    describe MultipartStreamUploader do
+      let(:client) { S3::Client.new(stub_responses: true) }
+      let(:subject) { MultipartStreamUploader.new(client: client) }
+      let(:params) { { bucket: 'bucket', key: 'key' } }
+      let(:one_mb) { '.' * 1024 * 1024 }
+      let(:seventeen_mb) { one_mb * 17 }
+
+      describe '#initialize' do
+        it 'constructs a default s3 client when none provided' do
+          client = double('client')
+          expect(S3::Client).to receive(:new).and_return(client)
+
+          uploader = MultipartStreamUploader.new
+          expect(uploader.client).to be(client)
+        end
+
+        it 'sets default configuration values when none provided' do
```

Comment:
Ditto for my previous comment - does not really test anything. If you want to test tempfile option does something, then assert one is made and used in an upload test. Assert part sizes and thread counts are what you expect in an upload test.